File: wiki/versions/1.19/entity/entity_attributes/scala/src/main/scala/org/quiltmc/wiki/entity_attributes/JumpStick.scala
package org.quiltmc.wiki.entity_attributes

import com.google.common.collect.{ImmutableMultimap, Multimap}
import net.minecraft.entity.EquipmentSlot
import net.minecraft.entity.attribute.{EntityAttribute, EntityAttributeModifier}
import net.minecraft.item.Item
import net.minecraft.item.Item.Settings

class JumpStick(settings: Settings) extends Item(settings) {

    override def getAttributeModifiers(slot: EquipmentSlot): Multimap[EntityAttribute, EntityAttributeModifier] =
        // If held in main hand, return our modifier, otherwise nothing
        if (slot == EquipmentSlot.MAINHAND) ImmutableMultimap.of(AttributesExample.GENERIC_JUMP_BOOST, AttributesExample.SOME_MODIFIER)
        else super.getAttributeModifiers(slot)
}

File: wiki/versions/1.19/entity/entity_attributes/kotlin/src/main/kotlin/org/quiltmc/wiki/entity_attributes/JumpStick.kt
package org.quiltmc.wiki.entity_attributes

import com.google.common.collect.ImmutableMultimap
import com.google.common.collect.Multimap
import net.minecraft.entity.EquipmentSlot
import net.minecraft.entity.attribute.EntityAttribute
import net.minecraft.entity.attribute.EntityAttributeModifier
import net.minecraft.item.Item

class JumpStick(settings: Settings): Item(settings) {

    override fun getAttributeModifiers(slot: EquipmentSlot): Multimap<EntityAttribute, EntityAttributeModifier> {
        // If held in main hand, return our modifier, otherwise nothing
        return if (slot == EquipmentSlot.MAINHAND) ImmutableMultimap.of(GENERIC_JUMP_BOOST, SOME_MODIFIER)
        else super.getAttributeModifiers(slot)
    }
}
